### ISSUE TITLE
Unexpected Cake\Filesystem\File::copy behavior

### ISSUE THREAD

**Issue by cnizzardini** [cakephp/cakephp#12567]
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.5.17
* Platform and Target: Ubuntu 18.04 LTS, Apache/2.4.33, PHP 7.0.31-1+ubuntu18.04.1+deb.sury.org+1

### What you did

I had expected a file copy to create a copy in the the files current directory, instead the copy was created at the webroot. While I have no problem specifying a full path, I find this to be odd behavior. Not sure if a change is worth it or even possible due to BC.

```php
$objFile = new File('/var/some/path/to/image.jpg');
$objFile->copy('image_copy.jpg');
```

### What happened
Copy was created at `webroot/`

### What you expected to happen
Copy to be created at same working directory as the original.

I don't actually expect this to be addressed because of BC, but at a minimum the documentation should specify File::copy() expects an absolute path as its argument.

**Comment by markstory** [comment]
I will take a look, but this looks like a scenario where File makes a bad assumption like you said. I think we may have corrected this class of issue in the upcoming 4.0 release. If we haven't it makes sense to either require an absolute path or use the current directory. I agree that this is something that is unlikely to change in 3.x for BC reasons.

**Comment by cnizzardini** [comment]
I expected as much, documentation should then specify absolute path.
